circus-train-avro/src/main/java/com/hotels/bdp/circustrain/avro/util/FileSystemPathResolver.java

```diff
@@ -58,21 +58,22 @@ public Path resolveScheme(Path path) {
   }
 
   public Path resolveNameServices(Path path) {
-    String nameService = configuration.get(DFSConfigKeys.DFS_NAMESERVICES);
-    if (isNotBlank(nameService)) {
-      URI uri = path.toUri();
-      String scheme = uri.getScheme();
-      String url = uri.getPath();
-      final String original = path.toString();
-      if (isBlank(scheme)) {
-        url = String.format("/%s%s", nameService, path);
-        path = new Path(url);
-      } else {
-        path = new Path(scheme, nameService, url);
+    if ("hdfs".equalsIgnoreCase(path.toUri().getScheme())) {
+      String nameService = configuration.get(DFSConfigKeys.DFS_NAMESERVICES);
+      if (isNotBlank(nameService)) {
+        URI uri = path.toUri();
```

Comment:
NABD but you could consider moving these out of the if statement and then re-using them in both the `if()` clause and here.